js/components/higher-order/holdOnClick.js

```diff
@@ -0,0 +1,125 @@
+var React = require('react');
+var CONSTANTS = require('../../constants/constants');
+
+/**
+ * Extends a button component with the ability to periodically call the onClick
+ * handler while the button is being held down. Allows this behavior with both regular
+ * mouse click as well as with the Space and Enter keys.
+ * @public
+ * @param {Component} ComposedComponent The React component to extend.
+ * The component must support adding handlers the following events: click, keydown, mousedown, mouseup
+ * @return {Component} A new component that supports holding the onClick handler
+ */
+var holdOnClick = function(ComposedComponent) {
+
+  var CLICK_HOLD_START_DELAY = 500;
+  var CLICK_HOLD_FREQUENCY = 100;
```

Comment:
Should we have these values be configurable?

Reply:
This is not currently a requirement so I think they'll be predefined for the first version. These values mimic the behavior of the browser's `keydown` event so that both mouse and keyboard controls behave consistently.

Reply:
Got it, sounds good. No need to feature creep